Fix the dune build

gasche opened the pull request:
I use the dune build to get Merlin support on the compiler codebase. See [HACKING.adoc: Using Merlin](https://github.com/ocaml/ocaml/blob/trunk/HACKING.adoc#using-merlin).

gasche committed:
dune: tell dune about asmcomp/polling.ml

xavierleroy commented:
The might be a nicer fix: add an attribute in stdlib/genlex.{ml,mli} that turns warning 3 off.  

Then there's no need ot change stdlib/dune, and my hack in stdlib/Compflags can also be removed.

xavierleroy committed:
stdlib/genlex.{ml,mli}: use attributes to turn deprecation alerts off

Instead of a stdlib/Compflags rule.
This ensures compatibility with the Dune build.

xavierleroy commented:
I took the liberty to push the suggested fix to your branch.  Please squash and merge if you're happy, or holler if you're not.

gasche commented:
Thanks for the change, I agree the warning is nicer.

(I mulled over changing `[@@@ocaml.warning "-3"]` to `[@@@ocaml.alert "-deprecated"]`, which is more readable, but in the end I went for the more standard thing we are already using in other parts of the build system.)
